warp10/src/main/sh/warp10.sh

```diff
@@ -525,133 +391,168 @@ start() {
   else
     echo "Repairing LevelDB..."
-    ${JAVACMD} -cp ${WARP10_JAR} -Dfile.encoding=UTF-8 io.warp10.standalone.WarpRepair ${LEVELDB_HOME}
+    ${JAVACMD} -cp "${WARP10_JAR}" -Dfile.encoding=UTF-8 io.warp10.standalone.WarpRepair "${LEVELDB_HOME}"
   fi
 }
 
 compact() {
-  #
-  # Make sure the caller is WARP10_USER
-  #
-  isUser ${WARP10_USER}
+  isWarp10User
 
   if isStarted; then
-    echo "Operation has been cancelled! - Warp 10 instance must be stopped for compaction"
-    exit 1
+    die "Operation has been cancelled! - Warp 10 instance must be stopped for compaction"
   else
     echo "Compacting LevelDB..."
-    ${JAVACMD} -cp ${WARP10_JAR} io.warp10.leveldb.WarpCompact ${LEVELDB_HOME} $1 $2
+    ${JAVACMD} -cp "${WARP10_JAR}" io.warp10.leveldb.WarpCompact "${LEVELDB_HOME}" "$1" "$2"
   fi
 }
 
+##
+## Initialize script
+##
+getWarp10Home
+WARP10_REVISION=@VERSION@
+WARP10_CONFIG_DIR=${WARP10_HOME}/etc/conf.d
+WARP10_JAR=${WARP10_HOME}/bin/warp10-${WARP10_REVISION}.jar
+WARP10_CLASS=io.warp10.standalone.Warp
+PID_FILE=${WARP10_HOME}/logs/warp10.pid
+getConfigFiles
+getJava
+
+
+JAVA_VERSION=$("${JAVACMD}" -XshowSettings:all -version 2>&1|grep 'java.version = '|sed -e 's/.* //' -e 's/^1\.//' -e 's/\..*//' -e 's/-.*//')
+if [ "${JAVA_VERSION}" -gt 8 ]; then
+  JAVA_OPTS="--add-exports java.desktop/com.sun.imageio.plugins.png=ALL-UNNAMED ${JAVA_OPTS:-}"
+fi
+
+##
+## Classpath
+## The lib directory is dedicated to user libraries (extensions, plugins...)
+##
+WARP10_CP=${WARP10_HOME}/etc:${WARP10_JAR}:${WARP10_HOME}/lib/*
+
+SENSISION_EVENTS_DIR=/var/run/sensision/metrics
+
+##
+## Set SENSISIONID as environment variable to identify this instance by Sensision
+##
+export SENSISIONID=warp10
+
+LOG4J_CONF=${WARP10_HOME}/etc/log4j.properties
+JAVA_HEAP_DUMP=${WARP10_HOME}/logs/java.heapdump
+# you can specialize your metrics for this instance of Warp10
+#SENSISION_DEFAULT_LABELS=-Dsensision.default.labels=instance=warp10-test,env=dev
+JAVA_OPTS="-Djava.awt.headless=true -Dlog4j.configuration=file:${LOG4J_CONF} -Dsensision.server.port=0 ${SENSISION_DEFAULT_LABELS:-} -Dsensision.events.dir=${SENSISION_EVENTS_DIR} -Dfile.encoding=UTF-8 -Xms${WARP10_HEAP} -Xmx${WARP10_HEAP_MAX} -XX:+UseG1GC ${JAVA_OPTS:-} ${JAVA_EXTRA_OPTS:-}"
+export MALLOC_ARENA_MAX=1
+
 # See how we were called.
 case "${1:-}" in
-  bootstrap)
-  bootstrap
+init)
+  case "${2:-}" in
+  distributed)
+    distConf
+    ;;
+  standalone)
+    leveldbConf
+    ;;
+  standalone+)
+    fdbConf
+    ;;
+  in-memory)
+    inmemoryConf
+    ;;
+  *)
+    die "Usage: $0 init <distributed|standalone|standalone+|in-memory>"
+    ;;
+  esac
   ;;
-  start)
+start)
   start
   ;;
-  jmxstart)
-  JAVA_OPTS="${JAVA_OPTS} -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=${JMX_PORT:-1098}"
-  echo "## WARNING: JMX is enabled on port ${JMX_PORT:-1098}"
+jmxstart)
```

Comment:
This should be removed, JMX will be enabled if the JMX_PORT environment variable is set.

Reply:
done